pkgs/tools/admin/certbot/default.nix

```diff
@@ -38,6 +40,25 @@ buildPythonApplication rec {
 
   checkInputs = [ pytest_xdist pytest dateutil ];
 
+  # certbot.withPlugins has a similar calling convention as python*.withPackages
+  # it gets invoked with a lambda, and invokes that lambda with the python package set matching certbot's:
+  # certbot.withPlugins (cp: [ cp.certbot-dns-foo ])
+  passthru.withPlugins = f: let
+    # call the lambda with python.pkgs, get back a list of plugins
+    plugins = (f python.pkgs);
+    # combine all required python modules
+    runtimeClosure = lib.flatten (map (p: [p] ++ p.passthru.requiredPythonModules) plugins);
```

Comment:
there is the function `requiredPythonModules` that does this. If plugins is not part of the Python packages set, it's IMO better to override the set and inject them.

Reply:
certbot plugins are python modules, so this is probably obsoleted.